src/sentry/integrations/slack/message_builder/issues.py

```diff
@@ -372,13 +373,52 @@ def get_action_text(text: str, actions: Sequence[Any], identity: RpcIdentity) ->
     return action_text
 
 
+def _ignore_action(group: Group) -> SlackMessageAction | None:
+    if group.issue_category == GroupCategory.FEEDBACK:
+        return None
+
+    if group.get_status() == GroupStatus.IGNORED:
+        return SlackMessageAction(
+            name="status", label="Mark as Ongoing", value="unresolved:ongoing"
+        )
+
+    return SlackMessageAction(name="status", label="Archive", value="archive_dialog")
+
+
+def _resolve_action(group: Group, project: Project) -> SlackMessageAction:
+    if group.get_status() == GroupStatus.RESOLVED:
+        return SlackMessageAction(
+            name="unresolved:ongoing", label="Unresolve", value="unresolved:ongoing"
+        )
+    if not project.flags.has_releases:
+        return SlackMessageAction(name="status", label="Resolve", value="resolved")
+
+    return SlackMessageAction(
+        name="status",
+        label="Resolve",
+        value="resolve_dialog",
+    )
+
+
+def _assign_action(group: Group) -> SlackMessageAction:
```

Comment:
moved these out from being anonymous closures in a function to named functions that we can now test and enforce contracts